src/bot/checks.rs

```diff
@@ -0,0 +1,137 @@
+use poise::serenity_prelude::{ChannelId, Permissions, RoleId};
+use sqlx::{MySql, Pool};
+use std::env;
+use tracing::error;
+
+use crate::bot::Context;
+use crate::mysql_lib;
+
+/// Checks if the author has any of the specified roles
+///
+/// Will return false when:
+/// - the message was not send in a guild
+/// - the author doesn't have any of the roles
+pub async fn has_one_of_roles(ctx: Context<'_>, roles: Vec<RoleId>) -> bool {
+    if let Some(member) = ctx.author_member().await {
+        return member.roles.iter().any(|role| roles.contains(role));
+    }
+    false
+}
+
+/// Checks if the author has the specified role
+///
+/// Will return false when:
+/// - the message was not send in a guild
+/// - the author doesn't the role
+#[allow(dead_code)]
+pub async fn has_role(ctx: Context<'_>, role: RoleId) -> bool {
+    has_one_of_roles(ctx, vec![role]).await
+}
+
+/// Checks that the author doesn't have any the specified role
+///
+/// Will return true when:
+/// - the message was not send in a guild
+/// - the author doesn't have any of the role
+#[allow(dead_code)]
+pub async fn has_not_roles(ctx: Context<'_>, roles: Vec<RoleId>) -> bool {
+    !has_one_of_roles(ctx, roles).await
+}
+
+/// Checks that the author doesn't have the specified role
+///
+/// Will return true when:
+/// - the message was not send in a guild
+/// - the author doesn't have the role
+#[allow(dead_code)]
+pub async fn has_not_role(ctx: Context<'_>, role: RoleId) -> bool {
+    has_not_roles(ctx, vec![role]).await
+}
+
+/// Checks if the author has a role that gives him admin permissions
+///
+/// Will return false when:
+/// - the message was not send in a guild
+/// - the author doesn't have a role with admin permissions
+#[allow(dead_code)]
+pub async fn is_admin(ctx: Context<'_>) -> bool {
+    if let Some(member) = ctx.author_member().await {
+        if let Some(guild) = ctx.guild() {
+            return member.roles.iter().any(|role_id| {
+                guild.roles.get(role_id).map_or(false, |role| {
+                    role.has_permission(Permissions::ADMINISTRATOR)
+                })
+            });
+        }
+    }
+    false
+}
+
+/// Checks if the author is the owner of the guild where the message was send
```

Comment:
Sorry if this is nit-picking but the last word should be "sent" (also in a couple other places)

Reply:
oh yes true